WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/CardReaderSettingsKnownReadersProvider.swift

```diff
@@ -0,0 +1,24 @@
+import Foundation
+import Combine
+
+/// Defines a protocol for wrappers around AppSettingsStore for "Known" Card Readers to confom to
+/// Allows us to support multiple implementations/dependency injection, and avoid introducing
+/// Combine into the Storage layer
+///
+protocol CardReaderSettingsKnownReadersProvider {
+
+    /// We can't use a @Published wrapper for knownReaders as part of a protocol declaration, but we can
+    /// require its synthesized methods explicitly
```

Comment:
This seems to be working but I don't think it's using the API as intended. The documentation for [Published](https://developer.apple.com/documentation/combine/published) says:

> The @Published attribute is class constrained. Use it with properties of classes, not with non-class types like structures

I understand that `@Published` is meant to be used on properties of an `ObservableObject`. The source for Combine isn't available, but looking at [OpenCombine](https://github.com/OpenCombine/OpenCombine/blob/master/Sources/OpenCombine/ObservableObject.swift#L52), an `ObservableObject` will use reflection to make sure every `@Published` property calls `objectWillChange` in the parent object.

If we don't need the `ObservableObject` features, I think a `CurrentValueSubject<[String], Never>` might do the same job with simpler code. We should probably expose it as a type-erased `AnyPublisher<[String], Never>`, since we don't want anything external sending values to the subject.

If we need outside access to both the publisher and the current value, then maybe we still need two properties.

I put up a quick diff to illustrate my suggestion:
https://gist.github.com/koke/80e760a5c76ac56080abbeb14437a711

Reply:
Thanks @koke - I think you're right. This could be a risky approach. We do need both the current value (for when the viewmodels initialize) and the publisher (to be notified of changes)

Reply:
Fixed in https://github.com/woocommerce/woocommerce-ios/pull/4517/commits/992361eaf8315e314396552dcbcc447a008679a3